pkg/apis/duck/v1/delivery_types.go

```diff
@@ -123,6 +128,17 @@ func (ds *DeliverySpec) Validate(ctx context.Context) *apis.FieldError {
 		}
 	}
 
+	if ds.Format != nil {
+		validFormats := map[string]bool{
+			"structured": true,
+			"binary":     true,
+			"ingress":    true,
+		}
+		if !validFormats[*ds.Format] {
+			errs = errs.Also(apis.ErrInvalidValue(*ds.Format, "format"))
```

Comment:
I would add the details of the error which the list of valid values (as 3rd parameter)